src/e2e/puppeteer/__tests__/gestures.ts

```diff
@@ -0,0 +1,24 @@
+import { KnownDevices } from 'puppeteer'
+import { drawHorizontalLineGesture } from '../helpers/gesture'
+import paste from '../helpers/paste'
+import { page } from '../setup'
+
+vi.setConfig({ testTimeout: 20000, hookTimeout: 20000 })
+const iPhone = KnownDevices['iPhone 15 Pro']
+
+describe('gestures', () => {
+  it.skip('when starting a gesture, the command palette will open', async () => {
+    const cursor = 'Hello'
+    const importText = `
+    - ${cursor}`
+
+    await page.emulate(iPhone)
```

Comment:
@raineorshine @trevinhofmann I cannot for the life of me get puppeteer to simulate a selection while emulating a mobile browser. I'll give it another try on Monday, but please let me know if you have any experience getting it to work.

Reply:
Thanks for trying! If it's difficult/impossible I'm okay skimping on test coverage here.

Reply:
Did you try creating a selection with touch events, or by directly accessing the browser selection API?

Reply:
I did try directly accessing the browser selection API, and found that it wouldn't hold onto the selection once it left the `page.evaluate` block. That might be worth another try, but I didn't have much luck the first time.

As for creating the selection with touch events, I tried to use a physical device to figure out how that would work, and it seemed like the only way to get a selection was to tap and then use the tooltip menu to "Select All". I don't think the tooltip menu appears in the Puppeteer emulator. If you know another, more reliable way to tap to create a a selection, please do let me know.

Reply:
A double tap selects the word under the touch. I imagine you will run into the same issue though in which the selection is lost after the `page.evaluate`.

We will have to rely on manual testing for this one 👍

Reply:
Double tap worked!

Reply:
I actually wasn't able to get the `browserless/chrome` v1 Docker image working on my Macbook M1 (presumably due to issues with the ARM chipset) so I've been running the v2 `ghcr.io/browserless/chromium` image. Maybe this is the first time that I've found a difference between the two versions. I'm planning to get a new computer next year, and I'll choose something with fewer Docker compatibility issues. That said, upgrading the pipeline to use the new image might also be a worthwhile project.